src/java.base/share/classes/java/lang/ModuleLayer.java

```diff
@@ -297,6 +299,37 @@ public Controller addOpens(Module source, String pn, Module target) {
             source.implAddOpens(pn, target);
             return this;
         }
+
+        /**
+         * Enables native access for a module in the layer if the caller's module
+         * has native access.
+         *
+         * <p> This method is <a href="foreign/package-summary.html#restricted"><em>restricted</em></a>.
+         * Restricted methods are unsafe, and, if used incorrectly, their use might crash
+         * the JVM or, worse, silently result in memory corruption. Thus, clients should refrain
+         * from depending on restricted methods, and use safe and supported functionalities,
+         * where possible.
+         *
+         * @param  target
+         *         The module to update
+         *
+         * @return This controller
+         *
+         * @throws IllegalArgumentException
+         *         If {@code target} is not in the module layer
+         *
+         * @throws IllegalCallerException
+         *         If the caller is in a module that does not have native access enabled
+         */
+         @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
+         @CallerSensitive
+         public Controller enableNativeAccess(Module target) {
+             ensureInLayer(target);
+             Reflection.ensureNativeAccess(Reflection.getCallerClass(), Module.class,
+                 "enableNativeAccess");
+             target.implAddEnableNativeAccess();
+             return this;
```

Comment:
ModuelLayer.enableNativeAccess looks fine, we iterated on that in panama-foreign/pull/729. I assume you'll add @since 20. Also you might want to check the alignment, it looks like the method is indented by 5 instead of the usual 4 spaces.